src/components/Wizard/Patterns/WizardPattern.js

```diff
@@ -0,0 +1,196 @@
+  onStepChanged,
+  onNext,
+  onBack,
+  nextStepDisabled,
+  title,
+  loadingTitle,
+  loadingMessage,
+  show,
+  onHide,
+  onExited,
+  stepButtonsDisabled,
+  cancelText,
+  backText,
+  nextText,
+  closeText,
+  loading
+}) => {
+  const onBackClick = () => {
+    goToStep(Math.max(activeStepIndex - 1, 0));
+  };
+
+  const onNextClick = () => {
+    goToStep(Math.min(activeStepIndex + 1, steps.length - 1));
+  };
+
+  const getActiveStep = (relativeToIndex = activeStepIndex) =>
+    steps[relativeToIndex];
+
+  const getPrevStep = (relativeToIndex = activeStepIndex) =>
+    relativeToIndex > 0 && steps[relativeToIndex - 1];
+
+  const getNextStep = (relativeToIndex = activeStepIndex) =>
+    relativeToIndex < steps.length - 1 && steps[relativeToIndex + 1];
+
+  const goToStep = newStepIndex => {
+    const currentStep = steps[activeStepIndex];
+    if (shouldPreventGoToStep(newStepIndex)) return;
+    if (onStepChanged) onStepChanged(newStepIndex);
+    if (newStepIndex === activeStepIndex + 1) {
+      if (currentStep.onNext) currentStep.onNext();
+      if (onNext) onNext(newStepIndex);
+    }
+    if (newStepIndex === activeStepIndex - 1) {
+      if (onBack) onBack(newStepIndex);
+    }
+  };
+
+  const shouldPreventGoToStep = newStepIndex => {
+    const targetStep = steps[newStepIndex];
+    const stepBeforeTarget = getPrevStep(newStepIndex);
+
+    const preventExitActive = steps[activeStepIndex].preventExit;
+    const preventEnterTarget = propExists(targetStep, 'preventEnter')
+      ? targetStep.preventEnter
+      : stepBeforeTarget && stepBeforeTarget.isInvalid;
+    const nextStepClicked = newStepIndex === activeStepIndex + 1;
+
+    return (
+      preventExitActive ||
+      preventEnterTarget ||
+      (nextStepClicked && nextStepDisabled)
+    );
+  };
+
+  const onFirstStep = activeStepIndex === 0;
+  const onFinalStep = activeStepIndex === steps.length - 1;
+  const activeStepStr = (activeStepIndex + 1).toString();
+  const activeStep = getActiveStep();
+  const prevStepUnreachable =
+    onFirstStep || activeStep.preventExit || getPrevStep().preventEnter;
+  const nextStepUnreachable =
+    nextStepDisabled ||
+    activeStep.isInvalid ||
+    activeStep.preventExit ||
+    getNextStep().preventEnter;
+
+  return (
+    <Modal
+      show={show}
+      onHide={onHide}
+      onExited={onExited}
+      dialogClassName="modal-lg wizard-pf"
+    >
+      <Wizard>
+        <Modal.Header>
+          <button
+            className="close"
+            onClick={onHide}
+            aria-hidden="true"
+            aria-label="Close"
+          >
+            <Icon type="pf" name="close" />
+          </button>
+          <Modal.Title>{title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body className="wizard-pf-body clearfix">
+          <WizardPatternBody
+            loadingTitle={loadingTitle}
+            loadingMessage={loadingMessage}
+            loading={loading}
+            steps={steps}
```

Comment:
future iteration... 👍

Reply:
I actually totally agree with that @vojtechszocs, now that I look at this stuff more. I think we should accept both, just like I accept `VerticalNavItem` children or an `items` prop in `VerticalNav`. Future PR I think.

Reply:
Exactly, it could be just another way to express items.

Just like `render()` in `VerticalNav`:

```js
const { items, children } = this.props;
// Nav items may be passed either as nested VerticalNavItem children, or as nested items in a prop.
// The items prop will take priority, if present, and must be an array of item objects (not React components).
// If the items prop is not present, items must be expressed as VerticalNavItem children instead.
```

Some time ago, we also briefly discussed the idea of "prop holder" React components - those which don't render anything but instead group all related data/callbacks. Prop holder components are to be interpreted (e.g. `React.Children.map`'ed to actual markup) by their owner component.

In other words, React components don't necessary have to represent visual elements of a page (even though most do). In the declarative `react-router`, for example, `<Router>` would be the owner and `<Route>` is the prop holder interpreted by it. (However, there's still a difference that router is an infrastructure non-visual component whereas e.g. vertical navigation is a visual element.)

So, you could make `<WizardPatternStep>` as well as `<VerticalNavItem>` to be prop holders, which means that item render stuff would be located in `<WizardPattern>` and `<VerticalNav>`.